modAionImpl/src/org/aion/zero/impl/blockchain/AionPendingStateImpl.java

```diff
@@ -226,14 +292,12 @@ private int getPendingTxSize() {
 
     @Override
     public synchronized List<AionTransaction> getPendingTransactions() {
-        return this.txPool.snapshot();
+            return this.txPool.snapshot();
     }
 
     public synchronized AionBlock getBestBlock() {
-        if (best == null) {
-            best = blockchain.getBestBlock();
-        }
-        return best;
+        best.set(blockchain.getBestBlock());
```

Comment:
Are there any implications of changing the logic from a lazy load to something that is grabbed every time?

Reply:
For now, only when Api create the block template will call this method. Inside the pendingState doesnt use this method anymore. the previous implement in the pendingState call this method anywhere.

replace to the Collections.synchronizedList (new ArrayList<>() )